rust/main/utils/run-locally/src/server.rs

```diff
@@ -0,0 +1,61 @@
+use std::io;
+
+use reqwest::Url;
+use serde::{Deserialize, Serialize};
+
+use crate::RELAYER_METRICS_PORT;
+
+/// Copied from agents/relayer/src/server/message_retry.rs
```

Comment:
can we not import this?

Reply:
It'd have to be moved to `hyperlane-base` to be imported in both usage sites, but this is a relayer-specific data structure so may not fit that well in `-base`. I think we have similar duplication between agent code and e2e elsewhere too, cleanest would be to define a new crate where these types are defined but I personally was ok with this small duplication (CI would fail if the structs diverged)

Reply:
remind me why we can't import from the relayer in run-locally?

Reply:
it looks like we already are actually, cc @kamiyaa 
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/99c91d8da34e417af536100cb4b726eae08b14c4/rust/main/utils/run-locally/Cargo.toml#L36

Reply:
Added! Good call!